client/src/features/session/components/StartNewSession.tsx

```diff
@@ -158,21 +158,32 @@ function BackButton() {
   const projectUrl = Url.get(Url.pages.project, projectUrlData);
 
   const location = useLocation<LocationState | undefined>();
-  const { from, filePath } = location.state ?? {};
-  const backUrl = from ?? projectUrl;
-  const backLabel =
-    from && filePath
-      ? `Back to ${filePath}`
-      : from
-      ? "Back to notebook file"
-      : `Back to ${pathWithNamespace}`;
-
-  return <GoBackButton label={backLabel} url={backUrl} />;
+  const { from, filePath, fromLanding } = location.state ?? {};
+  const buttonBackInfo = {
+    url: "",
+    label: "",
+  };
+  if (from && filePath) {
+    buttonBackInfo.label = `Back to ${filePath}`;
+    buttonBackInfo.url = from;
+  } else if (from) {
+    buttonBackInfo.label = "Back to notebook file";
+    buttonBackInfo.url = from;
+  } else if (fromLanding) {
+    buttonBackInfo.label = "Back to home";
+    buttonBackInfo.url = "/";
+  } else {
+    buttonBackInfo.label = `Back to ${pathWithNamespace}`;
+    buttonBackInfo.url = projectUrl;
+  }
+
+  return <GoBackButton label={buttonBackInfo.label} url={buttonBackInfo.url} />;
```

Comment:
Avoid mutations (re-assigning values):

```suggestion
  const backUrl = from ? from : fromLanding ? "/" : projectUrl;
  const backLabel =
    from && filePath
      ? `Back to ${filePath}`
      : from
      ? "Back to notebook file"
      : fromLanding
      ? "Back to home"
      : `Back to ${pathWithNamespace}`;

  return <GoBackButton label={backLabel} url={backUrl} />;
```